Crash on plan

### Terraform Version

Terraform v0.10.0

### Panic Output

I can't really share the crash.log as there are private info in it.


```
panic: runtime error: invalid memory address or nil pointer dereference
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23a327f]
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: 
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: goroutine 2567 [running]:
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsS3BucketRead(0xc4223e25b0, 0x299be00, 0xc42013a000, 0x0, 0x40045e0)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_s3_bucket.go:787 +0x1b3f
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc420cb3560, 0xc4223acb40, 0x299be00, 0xc42013a000, 0xc4207775c8, 0xc421b0be01, 0xc4201f9400)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:320 +0x21d
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc4217e6000, 0xc4223acaf0, 0xc4223acb40, 0x4a044b0, 0x0, 0x26a6d20)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:267 +0x91
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc42195eae0, 0xc4223ab560, 0xc4223abbb0, 0x0, 0x0)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /opt/teamcity-agent/work/222ea50a1b4f75f4/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:510 +0x4e
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: reflect.Value.call(0xc42038ef60, 0xc420130908, 0x13, 0x2c7773a, 0x4, 0xc42111af20, 0x3, 0x3, 0xf, 0x0, ...)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /usr/local/go/src/reflect/value.go:434 +0x91f
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: reflect.Value.Call(0xc42038ef60, 0xc420130908, 0x13, 0xc4208f1f20, 0x3, 0x3, 0x112d940, 0x3faf560, 0xc4223a56e0)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /usr/local/go/src/reflect/value.go:302 +0xa4
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: net/rpc.(*service).call(0xc421a02ec0, 0xc421a02e80, 0xc421a75848, 0xc420485600, 0xc421b273e0, 0x2648b60, 0xc4223ab560, 0x16, 0x2648ba0, 0xc4223abbb0, ...)
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /usr/local/go/src/net/rpc/server.go:387 +0x144
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4: created by net/rpc.(*Server).ServeCodec
2017/08/02 16:45:47 [DEBUG] plugin: terraform-provider-aws_v0.1.3_x4:   /usr/local/go/src/net/rpc/server.go:481 +0x404
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform init`
1. `terraform plan`

--- a/aws/resource_aws_s3_bucket.go
+++ b/aws/resource_aws_s3_bucket.go
@@ -784,21 +784,24 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 				rule["id"] = *lifecycleRule.ID
 			}
 			filter := lifecycleRule.Filter
-			if filter.And != nil {
-				// Prefix
-				if filter.And.Prefix != nil && *filter.And.Prefix != "" {
-					rule["prefix"] = *filter.And.Prefix
-				}
-				// Tag
-				if len(filter.And.Tags) > 0 {
-					rule["tags"] = tagsToMapS3(filter.And.Tags)
-				}
-			} else {
-				// Prefix
-				if filter.Prefix != nil && *filter.Prefix != "" {
-					rule["prefix"] = *filter.Prefix
+			if filter != nil {
+				if filter.And != nil {
+					// Prefix
+					if filter.And.Prefix != nil && *filter.And.Prefix != "" {
+						rule["prefix"] = *filter.And.Prefix
+					}
+					// Tag
+					if len(filter.And.Tags) > 0 {
+						rule["tags"] = tagsToMapS3(filter.And.Tags)
+					}
+				} else {
+					// Prefix
+					if filter.Prefix != nil && *filter.Prefix != "" {
+						rule["prefix"] = *filter.Prefix
+					}
 				}
 			}
+
 			// Enabled
 			if lifecycleRule.Status != nil {
 				if *lifecycleRule.Status == s3.ExpirationStatusEnabled {

